composer/trainer/trainer.py

```diff
@@ -1212,7 +1216,8 @@ def eval(self, is_batch: bool):
             for evaluator in self.state.evaluators:
                 dataloader = evaluator.dataloader.dataloader
                 metrics = self._ensure_metrics_device_and_dtype(evaluator.metrics)
-                if isinstance(dataloader.sampler, torch.utils.data.DistributedSampler):
+                if hasattr(dataloader, "sampler") and isinstance(dataloader.sampler,
```

Comment:
Same as above: Can you add a `# TODO` in the comments, saying that the hasattr check will be removed in #424?

Reply:
Will do.